src/bincode/optional.zig

```diff
@@ -0,0 +1,48 @@
+const std = @import("std");
+const sig = @import("../lib.zig");
+const bincode = sig.bincode;
+
+pub fn defaultToNullOnEof(comptime T: type, fields: struct {
+    free: ?fn (allocator: std.mem.Allocator, data: anytype) void = null,
+    hashmap: bincode.MaybeHashMapConfig(sig.utils.types.hashMapInfo(T)) = .{},
+}) bincode.FieldConfig(?T) {
+    const gen = struct {
```

Comment:
It's mostly muscle memory. The reason I default to this style is because structs which are not meant to be instantiated are supposed to be snake_case, as per the zig style guide, and the name `gen` is just an abbreviation for "generated", ie decls generated based on the comptime parameters/types of the parameters.
I'm fine with renaming these to `S` if you want.

Reply:
makes sense - would probs be better for the sake of consistency to follow the S pattern

Reply:
50e456e3b1989fcd0ba2797c2cd93e28d4d4384e